### PR TITLE
[Bugfix:Developer] Fix Doctrine debug mode type issues

### PR DESCRIPTION
**Pull request by williamjallen**
### What is the current behavior?
A PHP error currently occurs in some cases when Doctrine queries with DateTime objects are run in debug mode.  This behavior was likely introduced in https://github.com/Submitty/Submitty/pull/7295.

### What is the new behavior?
Resolves issue by formatting DateTime objects appropriately before displaying.  In addition, this PR fixes a secondary issue where the `$params` parameter in `formatQuery()` is null and thus unable to be iterated over.

### CODE REVIEW

**Review by MasterOdin — CHANGES_REQUESTED** [review]
Add testcases to https://github.com/Submitty/Submitty/blob/master/site/tests/app/libraries/database/DatabaseUtilsTester.php for the changed behavior.

File: site/app/libraries/database/DatabaseUtils.php [hunk 1]
```diff
@@ -4,9 +4,18 @@
 
 namespace app\libraries\database;
 
+use app\libraries\DateUtils;
+
 class DatabaseUtils {
     public static function formatQuery($sql, $params): string {
+        if (is_null($params)) {
```
Comment (MasterOdin) [inline site/app/libraries/database/DatabaseUtils.php:11]: How do we hit this case? When/how is `$params` null?

Comment (williamjallen) [reply site/app/libraries/database/DatabaseUtils.php:11]: I am unsure.  I fixed the issue with DateTime objects and another error about null `$params` popped up and I couldn't find a source of the issue so I just wrapped it in the `is_null` and moved on.

Comment (MasterOdin) [reply site/app/libraries/database/DatabaseUtils.php:11]: This was a thrown exception?

Comment (williamjallen) [reply site/app/libraries/database/DatabaseUtils.php:11]: This was the error: Untitled
(line 15 is the start of the for-loop in this example)

File: site/app/libraries/database/DatabaseUtils.php [hunk 2]
```diff
@@ -4,9 +4,18 @@
...
+            return '';
+        }
```
Comment (MasterOdin) [inline site/app/libraries/database/DatabaseUtils.php:13]: So given the above, let's make this change:

```suggestion
    public static function formatQuery(string $sql, ?array $params): string {
        if ($params === null) {
            return $sql;
        }
```

I forgot to add types in my original PR, so let's add them here, where this should be the expected shape given the bit about null. Also using `=== null` is a micro-optimization over `is_null`, but worth preferring as it avoids the overhead of a function call to do the exact same thing as equality. Finally, given null parameters, we should still have a string for the query to run, so should just return `$sql` as is.

**Review by MasterOdin — CHANGES_REQUESTED** [review]

**Comment by MasterOdin** [comment]
@williamjallen you still need to add testcases to https://github.com/Submitty/Submitty/blob/master/site/tests/app/libraries/database/DatabaseUtilsTester.php to cover the changes in behavior, good to merge after that.

**Comment by williamjallen** [comment]
I have some homework due tonight but I plan to have the tests finished by tomorrow

**Comment by MasterOdin** [comment]
@williamjallen Sorry for losing track on this, I made a commit https://github.com/Submitty/Submitty/pull/7354/commits/dd8c14e7b13fbfc36851e79f0d3b5320c744ca31 to add a missing test case (`null`), just something to consider for the next time tests are written to ensure full test coverage.
